src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1284,6 +1294,56 @@ private bool isFieldStatic(CORINFO_FIELD_STRUCT_* fldHnd)
             return HandleToObject(fldHnd).IsStatic;
         }
 
+        public void SetSequencePoints(IEnumerable<ILSequencePoint> ilSequencePoints)
+        {
+            Debug.Assert(ilSequencePoints != null);
+            Dictionary<int, SequencePoint> sequencePoints = new Dictionary<int, SequencePoint>();
+
+            foreach (var point in ilSequencePoints)
+            {
+                sequencePoints.Add(point.Offset, new SequencePoint() { Document = point.Document, LineNumber = point.LineNumber });
+            }
+
+            _sequencePoints = sequencePoints;
+        }
+
+        public void SetLocalVariables(IEnumerable<ILLocalVariable> localVariables)
+        {
+            Debug.Assert(localVariables != null);
+            HashSet<string> names = new HashSet<string>();
+            var localSlotToInfoMap = new Dictionary<int, ILLocalVariable>();
+
+            foreach (var v in localVariables)
+            {
+                ILLocalVariable modifiedLocal = v;
+                modifiedLocal.Name = _compilation.NameMangler.SanitizeName(modifiedLocal.Name);
+                if (!names.Add(v.Name))
+                {
+                    modifiedLocal.Name = string.Format("{0}_local{1}", v.Name, v.Slot);
+                    names.Add(modifiedLocal.Name);
+                }
+
+                localSlotToInfoMap[v.Slot] = modifiedLocal;
+            }
+
+            _localSlotToInfoMap = localSlotToInfoMap;
+        }
+
+        public void SetParameterNames(IEnumerable<string> parameters)
+        {
+            Debug.Assert(parameters != null);
+            var parameterIndexToNameMap = new Dictionary<int, string>();
+            int index = 0;
+
+            foreach (var p in parameters)
```

Comment:
Sorry, I don't get it. How do I use a for-loop on IEnumerable without knowledge of the underling containers?